File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.java
class Solution {
    public int[] topKFrequent(int[] nums, int k) {
        Map<Integer, Integer> map = new HashMap<>();
        
        for (int num : nums) {
            map.put(num, map.getOrDefault(num, 0) + 1);
        }
        
        PriorityQueue<Map.Entry<Integer, Integer>> queue = new PriorityQueue<>(Map.Entry.comparingByValue(Comparator.reverseOrder()));
        
        for (Map.Entry<Integer, Integer> set : map.entrySet()) {
            queue.add(set);
        }
        
        int[] res = new int[k];
        
        for (int i = 0; i < k; i++) {
            res[i] = queue.poll().getKey();
        }
        
        return res;
    }
}

File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.py
from queue import PriorityQueue
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        map = {}
        for num in nums:
            if num in map:
                map[num] = map[num] + 1
            else:
                map[num] = 1
        
        sorted_list = sorted(map.items(), key = lambda item: item[1], reverse = True)
        first_k = sorted_list[:k]
        res = [item[0] for item in first_k]
        
        return res
        